RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -995,19 +995,67 @@ reco::Track TrackExtenderWithMTDT<TrackCollection>::buildTrack(const reco::Track
     double thiterror = -1.;
     bool validmtd = false;
 
-    //need to better handle the cases with >1 hit in MTD
+    size_t ifirst(0), ihitcount(0), ietlcount(0);
     for (auto const& hit : trajWithMtd.measurements()) {
-      bool ismtd = hit.recHit()->geographicalId().det() == DetId::Forward &&
-                   ForwardSubdetector(hit.recHit()->geographicalId().subdetId()) == FastTime;
-      if (ismtd) {
-        const MTDTrackingRecHit* mtdhit = static_cast<const MTDTrackingRecHit*>(hit.recHit()->hit());
-        thit = mtdhit->time();
-        thiterror = mtdhit->timeError();
-        validmtd = true;
-        break;
+      if (hit.recHit()->geographicalId().det() == DetId::Forward &&
+          ForwardSubdetector(hit.recHit()->geographicalId().subdetId()) == FastTime) {
+        ihitcount++;
+        if (ihitcount == 1 && ifirst == 0) {
+          ifirst = ihitcount;
+        }
+        if (MTDDetId(hit.recHit()->geographicalId()).mtdSubDetector() == 2) {
+          ietlcount++;
+        }
       }
     }
 
+    auto ihit1 = trajWithMtd.measurements().cbegin() + ifirst - 1;
```

Comment:
`ihit1` is only used when ihitcount>0, i.e. when `ifirst` is always 1

Unless there was some mistake above when you assigned `ifirst` (should it be different for ETL and BTL, instead?), this could then become
```suggestion
    auto ihit1 = trajWithMtd.measurements().cbegin()
```

Reply:
this needs to be rewritten ```ifirst``` can be removed now, I was using for cross check before fixing the ordering of layers, but now this is done